core/meterer/meterer.go

```diff
@@ -82,87 +82,66 @@ func (m *Meterer) MeterRequest(ctx context.Context, header core.PaymentMetadata,
 		}
 	}
 
+	// TODO(hopeyen): each quorum can have different min num symbols; the returned symbolsCharged is only for used for metrics.
+	// for now we simply return the charge for quorum 0, as quorums are likely to share the same min num symbols
+	// we can make this more granular by adding metrics to the meterer later on
+	_, protocolConfig, err := params.GetQuorumConfigs(OnDemandQuorumID)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get on-demand quorum config: %w", err)
+	}
+	symbolsCharged := SymbolsCharged(numSymbols, protocolConfig.MinNumSymbols)
 	return symbolsCharged, nil
 }
 
-// ServeReservationRequest handles the rate limiting logic for incoming requests
-func (m *Meterer) ServeReservationRequest(ctx context.Context, params *PaymentVaultParams, header core.PaymentMetadata, reservations map[core.QuorumID]*core.ReservedPayment, symbolsCharged uint64, quorumNumbers []uint8, receivedAt time.Time) error {
-	m.logger.Info("Recording and validating reservation usage", "header", header, "reservation", reservations)
-
-	// Take all the quorumIDs from the reservations
-	quorumIDs := make([]core.QuorumID, 0, len(reservations))
-	reservationWindows := make(map[core.QuorumID]uint64, len(reservations))
-	requestReservationPeriods := make(map[core.QuorumID]uint64, len(reservations))
-	// Gather quorums the user had an reservations on and relevant quorum configurations
-	for quorumID := range reservations {
-		quorumIDs = append(quorumIDs, quorumID)
-		_, protocolConfig, err := params.GetQuorumConfigs(quorumID)
-		if err != nil {
-			return fmt.Errorf("failed to get protocol config for quorum %d: %w", quorumID, err)
-		}
-		reservationWindows[quorumID] = protocolConfig.ReservationRateLimitWindow
-		requestReservationPeriods[quorumID] = GetReservationPeriodByNanosecond(header.Timestamp, protocolConfig.ReservationRateLimitWindow)
-	}
-	// Validate quorumIDs is a subset of the quorumNumbers in the dispersal request; this should be guaranteed by GetReservedPaymentByAccountAndQuorums
-	if err := ValidateQuorum(quorumNumbers, quorumIDs); err != nil {
-		return fmt.Errorf("invalid quorum for reservation: %w", err)
-	}
-
-	// Validate the used reservations are active and is of valid periods
-	for quorumID, reservation := range reservations {
-		if !reservation.IsActiveByNanosecond(header.Timestamp) {
-			return fmt.Errorf("reservation not active")
-		}
-		if !ValidateReservationPeriod(reservation, requestReservationPeriods[quorumID], reservationWindows[quorumID], receivedAt) {
-			return fmt.Errorf("invalid reservation period for reservation on quorum %d", quorumID)
-		}
+// serveReservationRequest handles the rate limiting logic for incoming requests
+func (m *Meterer) serveReservationRequest(ctx context.Context, globalParams *PaymentVaultParams, header core.PaymentMetadata, reservations map[core.QuorumID]*core.ReservedPayment, numSymbols uint64, quorumNumbers []uint8, receivedAt time.Time) error {
```

Comment:
can we wrap to multiple lines for readability
```suggestion
func (m *Meterer) serveReservationRequest(
    ctx context.Context,
    globalParams *PaymentVaultParams,
    header core.PaymentMetadata,
    reservations map[core.QuorumID]*core.ReservedPayment,
    numSymbols uint64,
    quorumNumbers []uint8,
    receivedAt time.Time,
) error {
```